### PR TITLE
Tee reader problems

### PR DESCRIPTION
**Pull request by kacf**

### CODE REVIEW

File: src/common/io.hpp [hunk 1]
```diff
@@ -297,9 +297,9 @@ class BufferedReader : virtual public Reader {
 	bool rewind_consumed_ {false};
 	bool stop_done_ {false};
 
-protected:
```
Comment (lluiscampos) [inline src/common/io.hpp:300]: This was meant for testing - see the CI failure

**Review by lluiscampos — APPROVED** [review]
Hats off for finding this issue. The bug is pretty clear in hindsight.

File: src/common/io.hpp (hunk 1 above)
Comment (kacf) [reply src/common/io.hpp:300]: Right.

**Comment by kacf** [comment]
TeeReader is not used by the client, only by downstream projects, so merging without a full run.

**Pull request merged**